src/components/pages/not-found.jsx

```diff
@@ -1,31 +1,74 @@
-import {Link, locationShape} from "react-router"
+import { Link, locationShape, routerShape } from "react-router"
 import React, {PropTypes} from "react"
 import DocumentTitle from "react-document-title"
-
+import {FormattedMessage} from "react-intl"
 
 const NotFoundPage = React.createClass({
+  contextTypes: {
+    router: routerShape.isRequired
+  },
+
   propTypes: {
     location: locationShape.isRequired,
     message: PropTypes.string,
+    countryPackageName: PropTypes.string
+  },
+
+  handleInputChange(event) {
+    this.setState({inputValue: event.target.value})
   },
+
+  handleSubmit(event) {
+    event.preventDefault()
+    this.context.router.push({
+      query: {q: this.state.inputValue},
+      hash: `#search-input`,
+    })
+  },
+
   render() {
     const {pathname} = this.props.location
-    const message = this.props.message || `La page « ${pathname} » n'existe pas.`
+    const countryPackageName = this.props.countryPackageName
+    const changelogURL = `http://www.github.com/openfisca/${countryPackageName}/blob/master/CHANGELOG.md`
```

Comment:
Use `gitHubProject `, which is [set in the configuration](https://github.com/openfisca/legislation-explorer/blob/master/config/production.france.js#L8).